theano/tensor/var.py

```diff
@@ -154,6 +154,14 @@ def __sub__(self, other):
         except (NotImplementedError, AsTensorError):
             return NotImplemented
 
+    def __matmul__(self, other):
+        # See explanation in __add__ for the error catched
+        # and the return value in that case
+        try:
+            return theano.tensor.dot(self, other)
```

Comment:
I don't think this is right. I think numpy do something a little different. Can you add tests for many type of ndim inputs and compare that to that operator in numpy? The tests would run only in python 3.5.